diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -554,7 +554,15 @@ dnl --------------------------------------------------------------
 FP_ARG_WITH_PATH_GNU_PROG([RANLIB], [ranlib], [ranlib])
 RanlibCmd="$RANLIB"
 RANLIB="$RanlibCmd"
+AC_SUBST([RanlibCmd])
+
+dnl ** Which strip touse?
+dnl--------------------------------------------------------------
+FP_ARG_WITH_PATH_GNU_PROG([STRIP], [strip],[strip])
+StripCmd="$STRIP"
+STRIP="$StripCmd"
+AC_SUBST([StripCmd])
 
 
 # Note: we may not have objdump on OS X, and we only need it on Windows (for DLL checks)
 case $HostOS_CPP in
diff --git a/mk/config.mk.in b/mk/config.mk.in
--- a/mk/config.mk.in
+++ b/mk/config.mk.in
@@ -775,7 +775,7 @@
 # The hack of using `:` to disable stripping is implemented by ghc-cabal.
 STRIP_CMD      = :
 else
-STRIP_CMD      = strip
+STRIP_CMD      = @StripCmd@
 endif
 PATCH_CMD      = @PatchCmd@
 TAR_CMD        = @TarCmd@

