diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -556,6 +556,13 @@ RanlibCmd="$RANLIB"
 RANLIB="$RanlibCmd"
 AC_SUBST([RanlibCmd])
 
+dnl ** Which strip to use?
+dnl --------------------------------------------------------------
+FP_ARG_WITH_PATH_GNU_PROG([STRIP], [strip], [strip])
+StripCmd="$STRIP"
+STRIP="$StripCmd"
+AC_SUBST([StripCmd])
+
 
 # Note: we may not have objdump on OS X, and we only need it on Windows (for DLL checks)
 case $HostOS_CPP in
diff --git a/mk/config.mk.in b/mk/config.mk.in
index 4e61eea821..fcbc32d7df 100644
--- a/mk/config.mk.in
+++ b/mk/config.mk.in
@@ -785,7 +785,7 @@ else ifeq "$(TARGETPLATFORM)" "arm-unknown-linux"
 # The hack of using `:` to disable stripping is implemented by ghc-cabal.
 STRIP_CMD      = :
 else
-STRIP_CMD      = strip
+STRIP_CMD      = @StripCmd@
 endif
 PATCH_CMD      = @PatchCmd@
 TAR_CMD        = @TarCmd@
